server/websocket.go

```diff
@@ -812,6 +815,21 @@ func (s *Server) wsUpgrade(w http.ResponseWriter, r *http.Request) (*wsUpgradeRe
 				ws.cookieJwt = c.Value
 			}
 		}
+		if opts.Websocket.UsernameCookie != _EMPTY_ {
```

Comment:
Since it looks like we are doing it more than once (above with JWTCookie), maybe we could use `r.Cookies` to avoid parsing the headers multiple times. Something like:
```
if cookies := r.Cookies(); len(cookies) > 0 {
	ows := &opts.Websocket
	for _, c := range cookies {
		if ows.JWTCookie == c.Name {
			ws.cookieJwt = c.Value
		} else if ows.UsernameCookie == c.Name {
			ws.cookieUsername = c.Value
		} else if ows.PasswordCookie == c.Name {
			ws.cookiePassword = c.Value
		} else if ows.TokenCookie == c.Name {
			ws.cookieToken = c.Value
		}
	}
}
```